detector_test.go

```diff
@@ -58,3 +59,28 @@ func TestDetector(t *testing.T) {
 		}
 	}
 }
+
+func detectBest(d *chardet.Detector, b []byte) (r *chardet.Result, err error) {
+	var all []chardet.Result
+	if all, err = d.DetectAll(b); err == nil {
+		r = &all[0]
+	}
+	return
+}
+
+func BenchmarkDetectBest(b *testing.B) {
+	textDetector := chardet.NewTextDetector()
+	aaaa := bytes.Repeat([]byte("A"), 1024)
+	b.Run("old", func(b *testing.B) {
+		b.ReportAllocs()
+		for i := 0; i < b.N; i++ {
+			detectBest(textDetector, aaaa)
+		}
+	})
+	b.Run("new", func(b *testing.B) {
```

Comment:
Thanks for the benchmark comparison! To commit the PR, please clean up/remove old implementation here and only leave the benchmark code for the latest implementation.